src/v/cloud_storage/remote.h

```diff
@@ -263,14 +263,17 @@ class remote : public ss::peering_sharded_service<remote> {
     materialized_segments& materialized() { return *_materialized; }
 
     /// Add partition manifest tags (includes partition id)
-    static std::vector<s3::object_tag>
-    get_manifest_tags(const model::ntp& ntp, model::initial_revision_id rev);
+    static s3::object_tag_formatter make_partition_manifest_tags(
+      const model::ntp& ntp, model::initial_revision_id rev);
     /// Add topic manifest tags (no partition id)
-    static std::vector<s3::object_tag> get_manifest_tags(
+    static s3::object_tag_formatter make_topic_manifest_tags(
       const model::topic_namespace& ntp, model::initial_revision_id rev);
     /// Add segment level tags
-    static std::vector<s3::object_tag>
-    get_segment_tags(const model::ntp& ntp, model::initial_revision_id rev);
+    static s3::object_tag_formatter
+    make_segment_tags(const model::ntp& ntp, model::initial_revision_id rev);
+    /// Add tags for tx-manifest
+    static s3::object_tag_formatter make_tx_manifest_tags(
+      const model::ntp& ntp, model::initial_revision_id rev);
```

Comment:
Maybe I'm misinterpreting the suggestion, but I thought by having the formatter store the values, this would look something like:

```
struct partition_manifest_tags {
   ntp ntp;
   revision_id rev;
   sstring format() const {
     ... uses `ntp` and `rev` to generate tags
   }
};
```

That way we don't have the overhead of keeping around long-lived strings for the tag names (i.e. "rp-ns" and friends).

Reply:
Currently, the formatter stores a single string that can be extended by calling `add` method. The alternative for this would be to store the parameters and to build the string per request. But some of our parameters are strings (topic name, namespace) so it won't improve memory usage much.